packages/badge/src/badge.css

```diff
@@ -0,0 +1,139 @@
+
+:host {
+    display: inline-flex;
+    align-items: center;
+}
+
+/* disable :hover until Spectrum CSS disables hover upstream */
+
+:host(:hover) {
+    background-color: var(--spectrum-badge-m-textonly-background-color);
+    color: var(--spectrum-badge-m-texticon-text-color);
+}
+
+/* implement fixed placement styling until Spectrum CSS adds it upstream */
+
+:host([fixed='top']) {
+    --spectrum-badge-neutral-textonly-border-top-left-radius: 0;
+    --spectrum-badge-neutral-textonly-border-top-right-radius: 0;
+}
+
+:host([fixed='right']) {
+    --spectrum-badge-neutral-textonly-border-top-right-radius: 0;
+    --spectrum-badge-neutral-textonly-border-bottom-right-radius: 0;
+}
+
+:host([fixed='bottom']) {
+    --spectrum-badge-neutral-textonly-border-bottom-left-radius: 0;
+    --spectrum-badge-neutral-textonly-border-bottom-right-radius: 0;
+}
+
+:host([fixed='left']) {
+    --spectrum-badge-neutral-textonly-border-top-left-radius: 0;
+    --spectrum-badge-neutral-textonly-border-bottom-left-radius: 0;
+}
+
+/* cascade badge's size to its icon */
+
+:host([size='s']) {
+    --spectrum-icon-tshirt-size-height: var(
+        --spectrum-alias-workflow-icon-size-s
+    );
+    --spectrum-icon-tshirt-size-width: var(
+        --spectrum-alias-workflow-icon-size-s
+    );
+    --spectrum-ui-icon-tshirt-size-height: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-75
+    );
+    --spectrum-ui-icon-tshirt-size-width: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-75
+    );
+}
+
+:host([size='m']) {
+    --spectrum-icon-tshirt-size-height: var(
+        --spectrum-alias-workflow-icon-size-m
+    );
+    --spectrum-icon-tshirt-size-width: var(
+        --spectrum-alias-workflow-icon-size-m
+    );
+    --spectrum-ui-icon-tshirt-size-height: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-100
+    );
+    --spectrum-ui-icon-tshirt-size-width: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-100
+    );
+}
+
+:host([size='l']) {
+    --spectrum-icon-tshirt-size-height: var(
+        --spectrum-alias-workflow-icon-size-l
+    );
+    --spectrum-icon-tshirt-size-width: var(
+        --spectrum-alias-workflow-icon-size-l
+    );
+    --spectrum-ui-icon-tshirt-size-height: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-200
+    );
+    --spectrum-ui-icon-tshirt-size-width: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-200
+    );
+}
+
+:host([size='xl']) {
+    --spectrum-icon-tshirt-size-height: var(
+        --spectrum-alias-workflow-icon-size-xl
+    );
+    --spectrum-icon-tshirt-size-width: var(
+        --spectrum-alias-workflow-icon-size-xl
+    );
+    --spectrum-ui-icon-tshirt-size-height: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-300
+    );
+    --spectrum-ui-icon-tshirt-size-width: var(
+        --spectrum-alias-ui-icon-cornertriangle-size-300
+    );
+}
+
+/* don't shrink icons and ensure they're separated from labels */
+
+::slotted([slot='icon']) {
```

Comment:
Oh, are there no rules for icons in this for icons in the actual CSS. Hrm. We may need to run this by Spectrum CSS for future additions on their end. In the interim, this likely wants to be more like Action Button, which is a little more complex. We might not have all of the custom properties to power it 🤔

Reply:
https://github.com/adobe/spectrum-css/issues/1443